facebook/OpenBIC
# wc: mb: Support sensor reading

Summary:
- Support ADC sensor reading.
- Support TMP75 device sensor reading via i2c.
- Support SSD device temperature sensor reading via i2c.
- Support HSC device sensor reading via i2c.
- Support CPU/DDR device sensor reading via peci.
- Support PCH/DDR device sensor reading from me via i2c.
- Support INA230 device sensor reading via i2c.

Test Plan:
- Build code: PASS
- Check sensor reading via BMC command: Pass

Log:
- Mainboard sensor reading from BMC console
  ```
  root@bmc-oob:~# sensor-util mb
  mb:
  MB MB Inlet Temp             (0x1) :   31.00 C     | (ok)
  MB MB Outlet Temp            (0x2) :   31.00 C     | (ok)
  MB PCH Temp                  (0x4) :   42.00 C     | (ok)
  MB CPU Temp                  (0x5) :   86.00 C     | (ucr)
  MB SOC Therm Margin          (0x6) :   14.00 C     | (ok)
  MB CPU TjMax                 (0x7) :  100.00 C     | (ok)
  MB SSD0 Temp                 (0x8) :   37.00 C     | (ok)
  MB HSC Temp                  (0x9) :   35.00 C     | (ok)
  MB DIMMA Temp                (0xA) :   32.00 C     | (ok)
  MB DIMMC Temp                (0xB) :   31.00 C     | (ok)
  MB DIMME Temp                (0xC) :   32.00 C     | (ok)
  MB DIMMG Temp                (0xD) :   31.00 C     | (ok)
  MB VCCIN SPS Temp            (0xE) : NA | (na)
  MB FIVRA SPS Temp            (0xF) : NA | (na)
  MB EHV SPS Temp              (0x10) : NA | (na)
  MB VCCD SPS Temp             (0x11) : NA | (na)
  MB FAON SPS Temp             (0x12) : NA | (na)
  MB P12V_STBY Vol             (0x20) :   11.71 Volts | (ok)
  MB P3V_BAT Vol               (0x21) :    0.79 Volts | (ok)
  MB P3V3_STBY Vol             (0x22) :    3.28 Volts | (ok)
  MB P1V8_STBY Vol             (0x23) :    1.70 Volts | (ok)
  MB P1V05_PCH Vol             (0x24) :    1.00 Volts | (ok)
  MB P5V_STBY Vol              (0x25) :    4.77 Volts | (ok)
  MB P12V_DIMM Vol             (0x26) :   11.73 Volts | (ok)
  MB P1V2_STBY Vol             (0x27) :    1.19 Volts | (ok)
  MB P3V3_M2 Vol               (0x28) :    3.27 Volts | (ok)
  MB HSC Input Vol             (0x29) :   11.94 Volts | (ok)
  MB VCCIN VR Vol              (0x2A) : NA | (na)
  MB FIVRA VR Vol              (0x2B) : NA | (na)
  MB EHV VR Vol                (0x2C) : NA | (na)
  MB VCCD VR Vol               (0x2D) : NA | (na)
  MB FAON VR Vol               (0x2E) : NA | (na)
  MB HSC Output Cur            (0x40) :    3.88 Amps  | (ok)
  MB VCCIN VR Cur              (0x41) : NA | (na)
  MB FIVRA VR Cur              (0x42) : NA | (na)
  MB EHV VR Cur                (0x43) : NA | (na)
  MB VCCD VR Cur               (0x44) : NA | (na)
  MB VCCD VR Cur in            (0x45) : NA | (na)
  MB FAON VR Cur               (0x46) : NA | (na)
  MB CPU PWR                   (0x60) :   87.00 Watts | (ok)
  MB HSC Input Pwr             (0x61) :   42.00 Watts | (ok)
  MB DIMMA PMIC_Pout           (0x62) :    0.00 Watts | (ok)
  MB DIMMC PMIC_Pout           (0x63) :    0.12 Watts | (ok)
  MB DIMME PMIC_Pout           (0x64) :    0.00 Watts | (ok)
  MB DIMMG PMIC_Pout           (0x65) :    0.12 Watts | (ok)
  MB VCCIN VR Pout             (0x66) : NA | (na)
  MB FIVRA VR Pout             (0x67) : NA | (na)
  MB EHV VR Pout               (0x68) : NA | (na)
  MB VCCD VR Pout              (0x69) : NA | (na)
  MB FAON VR Pout              (0x6A) : NA | (na)
  ```

- I/O Module board sensor reading from BMC console
  ```
  root@bmc-oob:~# sensor-util iom
  iom:
  IOM Temp                     (0x3) :   28.00 C     | (ok)
  IOM INA Vol                  (0x2F) :   12.04 Volts | (ok)
  IOM INA Cur                  (0x47) :    0.07 Amps  | (ok)
  IOM INA PWR                  (0x6B) :    0.86 Watts | (ok)
  ```
